src/compliment/utils.clj

```diff
@@ -181,36 +185,37 @@ Note that should always have the same value, regardless of OS."
       (->> (for [^String file (all-files-on-classpath classpath)
                  :when (and (.endsWith file ".class") (not (.contains file "__"))
                             (not (.contains file "$")))]
-             (.. (if (.startsWith file File/separator)
-                   (.substring file 1) file)
+             (.. (ensure-no-leading-slash file)
                  (replace ".class" "")
                  ;; Address the issue #79 , on Windows, for prefix such
                  ;; as "java.util.", the list of candidates was empty.
                  (replace resource-separator ".")))
            (group-by #(subs % 0 (max (.indexOf ^String % ".") 0)))))))
 
-(defn namespaces-on-classpath
-  "Returns the list of all Clojure namespaces obtained by classpath scanning."
+(defn namespaces&files-on-classpath
+  "Returns a collection of maps (e.g. `{:ns-str \"some.ns\", :file \"some/ns.cljs\"}`) of all clj/cljc/cljs namespaces obtained by classpath scanning."
   []
   (let [classpath (classpath)]
-    (cache-last-result ::namespaces-on-classpath classpath
-      (set (for [^String file (all-files-on-classpath classpath)
-                 :when (and (.endsWith file ".clj")
-                            (not (.startsWith file "META-INF")))
-                 :let [[_ ^String nsname] (re-matches #"[^\w]?(.+)\.clj" file)]
-                 :when nsname]
-             (.. nsname (replace resource-separator ".") (replace "_" "-")))))))
+    (set (cache-last-result ::namespaces-on-classpath classpath
+           (for [^String file (all-files-on-classpath classpath)
```

Comment:
`file` doesn't need a type hint anymore. Besides, none of the code around used binding pairs alignment, so I'd rather keep it that way.